api/src/main/java/org/infinispan/api/reactive/KeyValueStore.java

```diff
@@ -0,0 +1,131 @@
+package org.infinispan.api.reactive;
+
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.api.Experimental;
+import org.infinispan.api.reactive.listener.KeyValueStoreListener;
+import org.infinispan.api.reactive.query.QueryRequest;
+import org.reactivestreams.Publisher;
+
+/**
+ * A Reactive Key Value Store provides a highly concurrent and distributed data structure, non blocking and using
+ * reactive streams.
+ * <p>
+ *
+ * </p>
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @see <a href="http://www.infinispan.org">Infinispan documentation</a>
+ * @since 10.0
+ */
+@Experimental
+public interface KeyValueStore<K, V> {
+
+   /**
+    * Get the value of the Key if such exists
+    * @param key
+    * @return the value
+    */
+   CompletionStage<V> get(K key);
+
+   /**
+    * Insert the key/value if such key does not exist
+    *
+    * @param key
+    * @param value
+    * @return Void
+    */
+   CompletionStage<Void> insert(K key, V value);
+
+   /**
+    * Save the key/value. If the key exists will replace the value
+    * @param id
+    * @param value
+    * @return Void
+    */
+   CompletionStage<Void> save(K id, V value);
```

Comment:
replace `id` => `key` to be consistent with the other methods?

Reply:
ok